### DISCUSSION TITLE
Channels blocking.

### DISCUSSION THREAD

**Discussion by ramzymbk** [ReVanced/revanced-discussions#472] [Question] [answered]
The ability to completely block a specific channel and all its content. There is a Firefox/Chrome addon that does the exact thing called [Blocktube](https://github.com/amitbl/blocktube)  but it only works on desktop versions.

**Comment by LisoUseInAIKyrios** [comment, marked as answer]
This can now be done using hide keyword content, under `ReVanced Settings -> Feed`.

Enter the channel name as it appears in the feed/search (do not use the @ username).

Channel blocking does not work for Shorts, since since most Shorts do not show the channel name and thus do not get blocked.

Due to how keyword filtering works, the feature may not work on tablet devices (tablet home/subscription/search results do not use litho components). The only work around is to use `Change form factor` and pick a phone or automotive device, and then keyword filtering works again.

**Reply by ShahbazNaqvi** [reply]
its working  perfectly , but what if i want to block multipul channels?

**Reply by LisoUseInAIKyrios** [reply]
type in multiple channels, separated by new lines.

**Reply by ShahbazNaqvi** [reply]
wow ! thanks for answering :)

**Comment by TecnikOfficial** [comment]
this feature is a hit or miss dont seem to work i tried adding specific channels (following instructions) restarted youtube but it doesnt block  ... for example blocking using handle would be great .. like [skytube](https://github.com/SkyTubeTeam/SkyTube) in this fake news spread era we definitely need this feature

**Reply by LisoUseInAIKyrios** [reply]
You must use the channel name as it appears on the videos in the feed/search results.

If it's not working, then you must have typed in the wrong channel title name or you used the @ name (which is not supported).  Pay attention to the spacing between words and enter the exact letter casing the channel title uses.  ie: `MrBeast` hides his videos but `mrbeast` and `Mr Beast` does not hide his videos.

**Reply by TecnikOfficial** [reply]
thanks for reply...yes i tried that we need to be quite accurate to block something using handle to block would be a better option but anyways the issue i was facing  was not case sensitive the commas were causing problem instead of using commas I used new line and it worked ... although it didnt blocked channel from appearing in search but it blocks video results of that channel in search thanks

**Comment by Bucketsmith** [comment]
Even with a single channel name, as it appears in the search results, it does not seem to actually filter.
I've updated to the latest suggested version, same behaviour.
The filtering does not seem to be applied at all.

**Reply by LisoUseInAIKyrios** [reply]
Make sure you are using the exact channel name with the correct spacing/casing as described [here](https://github.com/orgs/ReVanced/discussions/472#discussioncomment-9312947).  Be aware that Shorts cannot be hidden by channel name.

If it still does not work then it might be a new layout change and your account was chosen for an a/b trial of the change.

You might fix this by using a different YT account, or turning on version spoofing in General layout.

**Comment by GKid94** [comment]
Waiting for the block option too 🙏, Not interested videos keep showing in search results

**Comment by ADHDefy** [comment]
Would it be possible to incorporate the ability to block channels based on the channel ID, like blocktube does? (For instance, the channel Barstool Sports is `UCifWD4FBa4eaKK7HLF0PlTA`). This is a little more accurate, and potentially, you could include the ability to import a BlockTube list so we can take all of our existing blocked channels over to YT Revanced from the desktop.

**Reply by LisoUseInAIKyrios** [reply]
Try using a channel id in hide keyword content.

**Reply by ADHDefy** [reply]
Didn't work, unfortunately. I added the channel ID as a keyword, made sure to check "hide search results by keyword" and was still able to pull up the channel itself and videos by that channel by searching the channel name.

**Reply by LisoUseInAIKyrios** [reply]
Then the only way is to block by the channel name, or using some other text that appears on screen.

Also see https://github.com/ReVanced/revanced-patches/issues/3399#issuecomment-2210521175
Channel names can usually be blocked using dashes around the name (must use exact format described in that comment).  And then it blocks the channel but does not block video titles that contain the channel name.

**Comment by Baramanga** [comment]
Is there a plan to incorporate a quick option to the 3 dot selection menu to quickly add a channel name to the filtered keyword? Making it deep in the settings is kinda annoying

**Reply by LisoUseInAIKyrios** [reply]
Adding items to that menu is very difficult.

Soon the settings will have a search feature, including showing past searches which can work as a shortcut to open any setting.

**Reply by Baramanga** [reply]
what about in the channel itself? Right under the "Hide user under my channel"
